app/src/main/kotlin/com/wire/android/ui/markdown/MarkdownComposer.kt

```diff
@@ -318,11 +311,40 @@ fun appendLinksAndMentions(
                 }
         }
     }
-    annotatedString.append(append)
+    annotatedString.append(updatedAnnotatedString)
     return updatedMentions
 }
 
-private fun convertTypoGraphs(child: Text) = child.literal
+fun highlightText(nodeData: NodeData, text: String): AnnotatedString {
+    var highlightIndexes = emptyList<MatchQueryResult>()
+
+    if (nodeData.searchQuery.isNotBlank()) {
+        highlightIndexes = QueryMatchExtractor.extractQueryMatchIndexes(
+            matchText = nodeData.searchQuery,
+            text = text
+        )
+    }
+
+    return buildAnnotatedString {
+        append(text)
+        highlightIndexes
+            .forEach { highLightIndex ->
+                if (highLightIndex.endIndex <= length) {
+                    addStyle(
+                        style = SpanStyle(
+                            background = nodeData.colorScheme.highlight.copy(alpha = 0.5f),
```

Comment:
issue: all text highlight colors were updated to match designs in this PR: https://github.com/wireapp/wire-android/pull/2681
it should be:
```suggestion
                            background = nodeData.colorScheme.highlight,
                            color = nodeData.colorScheme.onHighlight,
```